elife/php-long-running-processes.sls

```diff
@@ -0,0 +1,47 @@
+# only supports 16.04!
+
+{% if pillar.elife.php.processes.enabled %}
+{% for name, configuration in pillar.elife.php.processes.configuration.items() %}
+{% set hyphenized = name | replace('_', '-') %}
+{% set service_name = salt['elife.project_name']() + '-' + hyphenized %}
+php-long-running-process-service-{{ hyphenized }}:
+    file.managed:
+        - name: /lib/systemd/system/{{ service_name }}@.service
+        - source: salt://elife/config/lib-systemd-system-php-service.service
+        - template: jinja
+        - context:
+            name: {{ hyphenized }}
+            folder: {{ configuration.folder }}
+            command: {{ configuration.command }}
+        - require:
+            - php
+            {% if configuration.get('require', None) -%}
+            - {{ configuration.get('require') }}
```

Comment:
Does this allow for only 1 requirement?

Reply:
Yes, usually this is composer install or the configuration file provisioning, but integrates the two files on a single state. A project can still create a `cmd.run` state to converge its disparate states into one, rather than propagating multiple state names to builder-private or its pillar.